starfish/core/spots/DecodeSpots/per_round_max_channel_decoder.py

```diff
@@ -0,0 +1,45 @@
+from starfish.core.codebook.codebook import Codebook
+from starfish.core.intensity_table.decoded_intensity_table import DecodedIntensityTable
+from starfish.core.intensity_table.intensity_table_coordinates import \
+    transfer_physical_coords_to_intensity_table
+from starfish.core.spots.DecodeSpots.trace_builders import build_spot_traces_exact_match
+from starfish.core.types import SpotFindingResults
+from ._base import DecodeSpotsAlgorithm
+
+
+class PerRoundMaxChannel(DecodeSpotsAlgorithm):
+    """
+    Decode spots by selecting the max-valued channel in each sequencing round.
+
+    Note that this assumes that the codebook contains only one "on" channel per sequencing round,
+    a common pattern in experiments that assign one fluorophore to each DNA nucleotide and
+    read DNA sequentially. It is also a characteristic of single-molecule FISH and RNAscope
+    codebooks.
+
+    Parameters
+    ----------
+    codebook : Codebook
+        Contains codes to decode IntensityTable
+
+    """
+
+    def __init__(self, codebook: Codebook):
+        self.codebook = codebook
+
+    def run(self, spots: SpotFindingResults, *args) -> DecodedIntensityTable:
+        """Decode spots by selecting the max-valued channel in each sequencing round
+
+        Parameters
+        ----------
+        spots: SpotFindingResults
+            A Dict of tile indices and their corresponding measured spots
+
+        Returns
+        -------
+        DecodedIntensityTable :
+            IntensityTable decoded and appended with Features.TARGET and Features.QUALITY values.
+
+        """
+        intensities = build_spot_traces_exact_match(spots)
+        transfer_physical_coords_to_intensity_table(intensity_table=intensities, spots=spots)
+        return self.codebook.decode_per_round_max(intensities)
```

Comment:
out of scope: I think `Codebook.decode_per_round_max` should be brought over to this module.  thoughts?

Reply:
Yea that's actually something I've been thinking during this refactor. Especially since any new decoding method shouldn't have to add code to the Codebook class, I can make that PR